### PR TITLE
add Shape layer

### PR DESCRIPTION
**Pull request by LeicongLi**
## What changes were proposed in this pull request?
1. Added Shape layer to support loading ONNX model
https://github.com/onnx/onnx/blob/master/docs/Operators.md#shape

2. Added bridge between python api and onnx package in scala.

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If it is possible, please attach a screenshot; otherwise, remove this)

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/onnx/Shape.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * @param ev
+ * @tparam T The numeric type in this module parameters
+ */
+class Shape[T: ClassTag](implicit ev: TensorNumeric[T]) extends TensorModule[T] {
+
+  override def updateOutput(input: Tensor[T]): Tensor[T] = {
+    val dimSize = input.nDimension()
+    output = Tensor(dimSize)
+    (1 to dimSize).foreach(i => {
+      output.setValue(i, input.size(i).toFloat.asInstanceOf[T])
```
Comment (i8run) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/onnx/Shape.scala]: Use `ev.fromType` instead the `.toFloat.asInstanceOf[T]`.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/onnx/Shape.scala [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    })
+    output
+  }
+
+  override def updateGradInput(input: Tensor[T], gradOutput: Tensor[T]): Tensor[T] = {
+    gradInput = gradOutput
```
Comment (i8run) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/onnx/Shape.scala]: So why the `gradInput` is the same as the `gradOutput`. If the input is a `Tensor(Array(4, 3, 224, 224))`, so what's the content and shape of the `gradOutput` ?

Comment (LeicongLi) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/onnx/Shape.scala]: The function of Shape layer is taking an tensor, and output its shape.
e.g. Input Tensor(Array(4, 3, 224, 224)) => Output Tensor(Array(4)), a one dimension array with four elements (4, 3, 224, 224).

Comment (i8run) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/onnx/Shape.scala]: I'm confusing about the shape of gradInput and gradOutput. If the input is the shape of [4, 3, 224, 224], the gradOutput's shape will be [1, 4] or [4]. So the questions is, what's the shape of gradInput ? Is it [4, 3, 224, 224] or [4] ?

Comment (LeicongLi) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/onnx/Shape.scala]: I see what you are pointing out, gradInput should have the same shape as input which is [4, 3, 224, 224].

Comment (LeicongLi) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/onnx/Shape.scala]: It is a constant layer, I dont think the input gradient should be changed.
And currently, I might not enable backforward on this op.

**Pull request merged**